lib/src/Uncertainty/Algorithm/WeightedExperiments/SmolyakExperiment.cxx

```diff
@@ -0,0 +1,348 @@
+      << " level=" << level_
+      << " collection=" << collection_;
+  return oss;
+}
+
+Bool SmolyakExperiment::hasUniformWeights() const
+{
+  return false;
+}
+
+/* Sample generation */
+Sample SmolyakExperiment::generate() const
+{
+  Point weights;
+  return generateWithWeights(weights);
+}
+
+/* Compute the minimum of a multiindex */
+UnsignedInteger SmolyakExperiment::indicesMinimum(const Indices & indices) const
+{
+  const UnsignedInteger dimension = indices.getSize();
+  UnsignedInteger multiindexMin = indices[0];
+  for (UnsignedInteger j = 1; j < dimension; ++ j) multiindexMin = std::min(indices[j], multiindexMin);
+  return multiindexMin;
+}
+
+/* Compute multiindex set for combination rule*/
+IndicesCollection SmolyakExperiment::computeCombination() const
+{
+  const UnsignedInteger dimension = collection_.getSize();
+  // Compute the maximum 1-norm of the multi-index (Gerstner & Griebel, 1998), page 215
+  const UnsignedInteger maximumSum = level_ + dimension - 1;
+  LOGDEBUG(OSS() << "  maximumSum = " << maximumSum);
+  // Create a multi-index set from norm 1
+  const LinearEnumerateFunction enumerateFunction(dimension);
+  // Compute the size
+  const UnsignedInteger cardinalMax = enumerateFunction.getStrataCumulatedCardinal(level_ - 1);
+  const UnsignedInteger cardinalMin = enumerateFunction.getStrataCumulatedCardinal(level_ - dimension - 1);
+  const UnsignedInteger combinationIndicesCollectionSize = cardinalMax - cardinalMin;
+  LOGDEBUG(OSS() << "  combinationIndicesCollectionSize = " << combinationIndicesCollectionSize);
+  // Fill the indices
+  IndicesCollection combinationIndicesCollection(combinationIndicesCollectionSize, dimension);
+  UnsignedInteger multiindexIndex = 0;
+  const UnsignedInteger strataIndexMin = (level_ < dimension) ? 0 : level_ - dimension;
+  LOGDEBUG(OSS() << "  strataIndexMin = " << strataIndexMin);
+  LOGDEBUG(OSS() << "  level_ = " << level_);
+  for (UnsignedInteger strataIndex = strataIndexMin; strataIndex < level_; ++ strataIndex)
+  {
+    LOGDEBUG(OSS() << "  strataIndex = " <<  strataIndex);
+    const UnsignedInteger strataCardinal = enumerateFunction.getStrataCardinal(strataIndex);
+    const UnsignedInteger cumulatedCardinal = enumerateFunction.getStrataCumulatedCardinal(
+          strataIndex
+        );
+    const UnsignedInteger indexStart = cumulatedCardinal - strataCardinal;
+    for (UnsignedInteger i = indexStart; i < cumulatedCardinal; ++ i)
+    {
+      Indices indices(enumerateFunction(i));
+      for (UnsignedInteger j = 0; j < dimension; ++ j) indices[j] += 1;
+      LOGDEBUG(OSS() << "  indices = " <<  indices);
+      std::copy(indices.begin(), indices.end(), combinationIndicesCollection.begin_at(multiindexIndex));
+      ++ multiindexIndex;
+    } // loop over the indices in the strata
+  } // loop over the strata
+  return combinationIndicesCollection;
+}
+
+/* Comparison class with std:map interface. */
+class PointApproximateComparison
```

Comment:
Why is it confined in this class? IMO it should be a service of Point

Reply:
I disagree, this does not seem to be needed anywhere else, we want to really be sure before adding new methods to the base objects

Reply:
It could be used in UserDefined::compactSupport instead of the current implementation based on a rounding of the components